alchemy/src/neon/role.ts

```diff
@@ -0,0 +1,323 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import type { Secret } from "../secret.ts";
+import { handleApiError } from "./api-error.ts";
+import { createNeonApi, type NeonApiOptions, type NeonApi } from "./api.ts";
+
+/**
+ * Properties for creating or updating a Neon role
+ */
+export interface NeonRoleProps extends NeonApiOptions {
+  /**
+   * The ID of the project containing the branch
+   */
+  project_id: string;
+
+  /**
+   * The ID of the branch where the role will be created
+   */
+  branch_id: string;
+
+  /**
+   * Name of the role
+   */
+  name: string;
+
+  /**
+   * Whether to adopt an existing role if it already exists
+   * @default false
+   */
+  adopt?: boolean;
+}
+
+/**
+ * Response structure for role API operations
+ */
+export interface NeonRoleType {
```

Comment:
This is an internal type that is not needed outside of this file. Inline if possible. Move to bottom where the utility api functions are otherwise.